addons/crm/models/crm_lead.py

```diff
@@ -527,6 +530,59 @@ def _compute_ribbon_message(self):
             else:
                 lead.ribbon_message = False
 
+    @api.depends('email_from', 'partner_id', 'contact_name', 'partner_name')
+    def _compute_potential_lead_duplicates(self):
+
+        MIN_EMAIL_LENGTH = 7
+        MIN_NAME_LENGTH = 6
+        SEARCH_RESULT_LIMIT = 21
+
+        def return_if_relevant(model_name, domain):
+            # include archived records and transcend multi-company record rules
+            model = self.env[model_name].with_context(active_test=False)
```

Comment:
After discussing with Lna, we choose to finally to with sudo . Sorry for back and forth, simply re-add the sudo back. Idea is that it is better to have a counter and maybe see only part of it, and let people see with their manager. Could be added a docstring / comment / in commit message probably.